### PR TITLE
OHRM5X-1211:  maintenance bug fixes

### PR DESCRIPTION
**Pull request by Ajiharan007**
this PR contains
OHRM-1208: Supervisor is displayed as both the supervisor and the subordinate in the JSON file
OHRM-1207: Field names with no values are displayed in the JSON file
OHRM-1209: The date the user name was created and modified is not displayed in the JSON file.
OHRM-1210: "Past Employee" suffix is not displayed in the exported file name of a past employee
OHRM-1211:  "initializer ", "cloner" and "initialized" fields are displayed in the salary details of the downloaded file
OHRM-1212: 500 Error occurs when a user tries to download details of an employee with special characters in the employee name.
Middle Name placeholder props name and type changed [FE]

### CODE REVIEW

File: symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php [hunk 1]
```diff
@@ -32,21 +33,35 @@ abstract class DownloadFormat
      */
     abstract public function getFormattedString($values);
 
+    /**
+     * @param $fileName
+     * @return string
+     */
+    public function removeNonAsciiCharacters($fileName): string
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php:40]: example?

File: symfony/plugins/orangehrmMaintenancePlugin/test/AccessStrategy/downloadFormat/DownloadFormatTest.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public function testDownloadFileName()
+    {
+        $fileName=$this->jsonDownloadFormat->getDownloadFileName(1);
+        $this->assertEquals('Kayla T Abbey.json', $fileName);
+
+        $fileName=$this->jsonDownloadFormat->getDownloadFileName(2);
+        $this->assertEquals('Ashley$ ST Abel (Past Employee).json', $fileName);
+    }
+
+    public function testRemoveNonAsciiCharacters()
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmMaintenancePlugin/test/AccessStrategy/downloadFormat/DownloadFormatTest.php:48]:
```suggestion
    public function testRemoveNonAsciiCharacters(): void
```

File: symfony/plugins/orangehrmMaintenancePlugin/test/AccessStrategy/downloadFormat/DownloadFormatTest.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    protected JsonDownloadFormat $jsonDownloadFormat;
+
+    protected function setUp(): void
+    {
+        $this->jsonDownloadFormat=new JsonDownloadFormat();
+        $this->fixture = Config::get(Config::PLUGINS_DIR) . '/orangehrmMaintenancePlugin/test/fixtures/EmployeeDownloadFormat.yml';
+        TestDataService::populate($this->fixture);
+    }
+
+    public function testDownloadFileName()
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmMaintenancePlugin/test/AccessStrategy/downloadFormat/DownloadFormatTest.php:39]:
```suggestion
    public function testDownloadFileName(): void
```

File: symfony/plugins/orangehrmPimPlugin/entity/Decorator/EmployeeDecorator.php [hunk 4]
```diff
@@ -210,6 +210,6 @@ public function getFirstAndLastNames(): string
      */
     public function getFullName(): string
     {
-        return $this->getEmployee()->getFirstName() . ' ' . $this->getEmployee()->getMiddleName() . ' ' . $this->getEmployee()->getLastName();
+        return trim($this->getEmployee()->getFirstName()) . ' ' . trim($this->getEmployee()->getMiddleName()) . ' ' . trim($this->getEmployee()->getLastName());
     }
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/entity/Decorator/EmployeeDecorator.php:214]: Shall we add Employee entity test and cover this?

File: symfony/plugins/orangehrmAdminPlugin/Api/UserAPI.php [hunk 5]
```diff
@@ -136,6 +140,8 @@ public function create(): EndpointResourceResult
     {
         $user = new User();
         $this->setUserParams($user);
+        $user->setDateEntered($this->getDateTimeHelper()->getNow());
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAdminPlugin/Api/UserAPI.php:143]: we may have to add new test for this API class

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php (hunk 1 above)
Comment (Ajiharan007) [reply symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php:40]: test£$&*().json  replaced as test$&*().json

Comment (Ajiharan007) [reply symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php:40]: I found in HeaderUtil.php inbuilt class
 ```
 // filenameFallback is not ASCII.
        if (!preg_match('/^[\x20-\x7e]*$/', $filenameFallback)) {
            throw new \InvalidArgumentException('The filename fallback must only contain ASCII characters.');
        }
```

Comment (Ajiharan007) [reply symfony/plugins/orangehrmMaintenancePlugin/DownloadFormats/DownloadFormat.php:40]: if the filenameFallback is empty then it's throwing an error?
but just add string value to filenameFallback then it's not throw error but I was not sure it's the best approach

**Review by SharukaNP — COMMENTED** [review]
Good to me

File: symfony/plugins/orangehrmCorePlugin/Controller/AbstractFileController.php [hunk 6]
```diff
@@ -57,10 +57,8 @@ protected function getFileResponse(string $file): BinaryFileResponse
      */
     protected function makeAttachmentDisposition(string $filename): string
     {
-        return HeaderUtils::makeDisposition(
-            HeaderUtils::DISPOSITION_ATTACHMENT,
-            $filename
-        );
+        $params['filename*'] = "utf-8''" . rawurlencode($filename);
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Controller/AbstractFileController.php:60]: can you test with `/` & `\` chars

File: symfony/plugins/orangehrmAdminPlugin/Api/UserAPI.php (hunk 5 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmAdminPlugin/Api/UserAPI.php:143]: @devishke-orange can you help him to address this

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**